### PR TITLE
[PROD-2476] cover url enhancement

### PR DESCRIPTION
**Pull request by chetansatasiya**
### General Note
Keep all conversations related to this PR in the associated Jira issue(s). Do NOT add comment on this PR or edit this PR’s description.

### Notes to Developer
* Ensure the IDs (i.e. PROD-1) of all associated Jira issues are reference in this PR’s title
* Ensure that you have achieved the Definition of Done before submitting for review
* When this PR is ready for review, move the associate Jira issue(s) to “Needs Review” (or “Code Review” for Dev Tasks)

### Notes to Reviewer
* Ensure that the Definition of Done have been achieved before approving a PR
* When this PR is approved, move the associated Jira issue(s) to “Needs QA” (or “Approved” for Dev Tasks)

### CODE REVIEW

File: src/bp-core/bp-core-attachments.php [hunk 1]
```diff
@@ -1642,7 +1646,29 @@ function bp_attachments_cover_image_ajax_upload() {
...
+	 * Filters groups/members cover image attachment URL.
+	 *
+	 * @since BuddyBoss [BBVERSION]
+	 *
+	 * @param string $cover_url URL to the image.
+	 * @param array  $cover     Cover array.
+	 * @param string $component Component either groups or members.
+	 * @param int    $item_id   Inform about the item id the cover image was set for either group id or member id.
+	 */
+	$cover_url = apply_filters( 'bp_' . ( 'xprofile' === $component ) ? 'members' : $component . '_attachments_cover_image_url', $cover_url, $cover, ( 'xprofile' === $component ) ? 'members' : $component, $item_id );
```
Comment (eriktorsner) [inline src/bp-core/bp-core-attachments.php:1671]: Looks like a broken ternary operation. The ternary expression that calculates the filter name will always just return 'member' and nothing else, it needs to be wrapped in paratheses. See this snippet: https://sandbox.onlinephpfunctions.com/c/87b8d

**Review by eriktorsner — CHANGES_REQUESTED** [review]
Erik from DeliciousBrains here. See my comment in the code.

Also note that we haven't implemented the `bp_members_attachments_cover_image_url` in our integration. We're using `bp_attachments_pre_get_attachment` to rewrite cover image URLs at the moment but should probably also implement the one you are using here.

So while this PR won't get rid of the issue we originally reported (164603), it's a right step along the way.

File: src/bp-core/bp-core-attachments.php (hunk 1 above)
Comment (chetansatasiya) [reply src/bp-core/bp-core-attachments.php]: @eriktorsner fixed now

**Review by eriktorsner — APPROVED** [review]
Looking good to me 👍

**Review by KartikSuthar — APPROVED** [review]

**Pull request merged**